src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2667,7 +2662,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 			}
 		}
 
-		if ($this->blnHideFiles)
+		if (!$this->blnFiles && !$this->blnFilesOnly)
```

Comment:
Yes it does, because `$blnFiles` is [initialized to `true`](https://github.com/aschempp/contao-core-bundle/blob/6a7d052a4b08047e4d42cd9f084d27d195aea0b1/src/Resources/contao/drivers/DC_Folder.php#L84) and only [set to `false` on `initPicker`](https://github.com/aschempp/contao-core-bundle/blob/6a7d052a4b08047e4d42cd9f084d27d195aea0b1/src/Resources/contao/drivers/DC_Folder.php#L3046).